components/ai_chat/resources/page/state/data-context-provider.tsx

```diff
@@ -112,11 +113,29 @@ function DataContextProvider (props: DataContextProviderProps) {
     setHasUserDissmisedPremiumPrompt(true)
   }
 
-  const initialiseForTargetTab = () => {
+  const switchToDefaultModel = () => {
+    // Select the first non-premium model
+    const nonPremium = allModels.find(m => !m.isPremium)
+    if (!nonPremium) {
+      console.error('Could not find a non-premium model!')
+      return
+    }
+    setCurrentModel(nonPremium)
+  }
+
+  const updateCurrentPremiumStatus = async () => {
+    console.debug('Getting premium status...')
+    const premiumStatus = await getPageHandlerInstance().pageHandler.getPremiumStatus()
+    console.debug('got premium status: ', premiumStatus.result)
```

Comment:
Double checking - do you mean to leave these debug statements?

Reply:
yes, at least for now, so we can see 1) how long they're taking as I want to see if we can reproduce that http-request-every-time issue, and 2) how often it's asking since I'm not exactly sure when focus or visibility change